### PR TITLE
flashgbx: init at 3.37

### PR DESCRIPTION
**Pull request by grahamnorris**
## Description of changes

This PR adds [FlashGBX](https://github.com/lesserkuma/FlashGBX), a command-line and GUI utility for reading and writing to Game Boy, Game Boy Color, and Game Boy Advance cartridges using [GBxCart RW](https://www.gbxcart.com/) hardware.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Tested that this launches on aarch64-darwin and launches and functions correctly on x86_64-linux, including wayland.
The desktop icon does not appear correct on KDE plasma wayland, but AFAICT, this is due to the presence of a PNG rather than SVG icon.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/fl/flashgbx/package.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      exec = "flashgbx";
+      comment = "UI for reading and writing Game Boy and Game Boy Advance cartridges";
+      categories = [ "Utility" ];
+    };
+  in builtins.map makeDesktopItem [
+    common
+    (common // {
+      name = "${common.name}";
+      exec = "${common.exec}";
+    })
```
Comment (lolbinarycat) [inline pkgs/by-name/fl/flashgbx/package.nix:34]: it's unclear to me what exactly this is doing.  as far as i can tell it is overriding `common.name` with `common.name`, which seems like it wouldn't do anything?

Comment (grahamnorris) [reply pkgs/by-name/fl/flashgbx/package.nix]: Thanks for pointing this out. This was a remnant from the example I started from. I have now looked at a few other examples of `makeDesktopItem` and adapted the code, removing the mapping.

File: pkgs/by-name/fl/flashgbx/package.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "flashgbx";
+  version = "3.36";
+
+  src = fetchFromGitHub {
+    repo = "FlashGBX";
+    owner = "lesserkuma";
+    rev = "${version}";
```
Comment (lolbinarycat) [inline pkgs/by-name/fl/flashgbx/package.nix:16]:
```suggestion
    rev = version;
```

there's no need to interpolate here

**Review by lolbinarycat — CHANGES_REQUESTED** [review]

File: pkgs/by-name/fl/flashgbx/package.nix (hunk 2 above)
Comment (grahamnorris) [reply pkgs/by-name/fl/flashgbx/package.nix]: Another good catch, thanks.

File: pkgs/by-name/fl/flashgbx/package.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+python3Packages.buildPythonApplication rec {
+  pname = "flashgbx";
+  version = "3.36";
+
+  src = fetchFromGitHub {
+    repo = "FlashGBX";
+    owner = "lesserkuma";
+    rev = version;
+    sha256 = "sha256-MbeQKdmAGt4bTSleU1Dlve7fLDfldS2gIYUCIbgDAk4=";
```
Comment (lolbinarycat) [inline pkgs/by-name/fl/flashgbx/package.nix:17]:
```suggestion
    hash = "sha256-MbeQKdmAGt4bTSleU1Dlve7fLDfldS2gIYUCIbgDAk4=";
```

this is a small nitpick, but i'm pretty sure the `hash` attribute is preferred these days.

Comment (grahamnorris) [reply pkgs/by-name/fl/flashgbx/package.nix]: Okay, I'm happy to use the preferred nomenclature.

**Comment by rafameou** [comment]
Result of `nixpkgs-review pr 291763` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>flashgbx</li>
    <li>flashgbx.dist</li>
  </ul>
</details>

File: pkgs/by-name/fl/flashgbx/package.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      categories = [ "Utility" ];
+    })
+  ];
+
+  postInstall =
+  ''
+      install -D FlashGBX/res/icon.png $out/share/icons/hicolor/256x256/apps/flashgbx.png
+  '';
+
+  format = "pyproject";
```
Comment (rafameou) [inline pkgs/by-name/fl/flashgbx/package.nix:36]:
```suggestion
  pyproject = true;
```
https://github.com/NixOS/nixpkgs/issues/253154#issuecomment-1712673789

**Comment by rafameou** [comment]
I would also suggest changing the title of the pull request to "flashgbx: init at <version>"

File: pkgs/by-name/fl/flashgbx/package.nix (hunk 4 above)
Comment (grahamnorris) [reply pkgs/by-name/fl/flashgbx/package.nix]: Okay, thanks for this and the reference. I have addressed this in the latest push.

**Comment by rafameou** [comment]
Result of `nixpkgs-review pr 291763` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>flashgbx</li>
    <li>flashgbx.dist</li>
  </ul>
</details>

**Review by rafameou — APPROVED** [review]
LGTM, I can't test completely since I don't have the hardware, but overall it seems to be working well.

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 291763` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>flashgbx</li>
    <li>flashgbx.dist</li>
  </ul>
</details>

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [ ] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored


please add `setuptools` to nativeBuildInputs, and please swap the order of the commits

**Comment by grahamnorris** [comment]
Thanks for the review. I have addressed the comments.

I am not fully familiar with the nuances between the different `*Inputs` for a nix packages, but without `setuptools` also in `propagatedBuildInputs`, I only got the CLI version of the application instead of the GUI version on aarch64-darwin, so it seems the package has a run-time dependency on `setuptools`.

File: pkgs/by-name/fl/flashgbx/package.nix [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+  postInstall =
+  ''
+      install -D FlashGBX/res/icon.png $out/share/icons/hicolor/256x256/apps/flashgbx.png
+  '';
+
+  pyproject = true;
+
+  nativeBuildInputs = [
+    python3Packages.setuptools
```
Comment (pbsds) [inline pkgs/by-name/fl/flashgbx/package.nix:39]:
```suggestion
    python3Packages.setuptools
    copyDesktopItems
```

**Comment by pbsds** [comment]
the difference between nativeBuildInputs and buildInputs only matters for cross builds, where e.g. a x86 machine builds a ARM package. nativeBuildInputs would then be the x86 versions which can run on the host during build, while buildInputs would be ARM packages that get linked and propagated into the final build artifact. nativeBuildInputs get added to PATH, while buildInputs are added to configure scripts and such.

**Pull request merged**